### ISSUE TITLE
`use-maxsplit-arg` false positive when index is incremented inside a loop

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#4664]
### Steps to reproduce

Given a file `a.py`:
```python
source = 'A.B.C.D.E.F.G'
other_iterable = range(5)
i = 0
for j in other_iterable:
    print(source.split('.')[i])
    i += 1
```

### Current behavior

Result of `pylint a.py`:
```
split.py:5: [C0207(use-maxsplit-arg), ] Use source.split('.', maxsplit=1)[0] instead
```

### Expected behavior
no messages, since maxsplit=1 produces a different result.

### pylint --version output
```
pylint 2.9.3
astroid 2.6.2
Python 3.8.10 (v3.8.10:3d8993a744, May  3 2021, 08:55:58) 
```

**Comment by yushao2** [comment]
Thanks for reporting this! I'm able to reproduce this issue
